connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp

```diff
@@ -0,0 +1,1720 @@
+            _mod_params.params.lora.bandwidth = (lora_bandwidths_t) lora_bandwidths[bandwidth];
+            _mod_params.params.lora.coding_rate =
+                (lora_coding_states_t) coderate;
+
+            if (((bandwidth == 0) && ((datarate == 11) || (datarate == 12)))
+                || ((bandwidth == 1) && (datarate == 12)))
+            {
+                _mod_params.params.lora.low_datarate_optimization = 0x01;
+            }
+            else
+            {
+                _mod_params.params.lora.low_datarate_optimization = 0x00;
+            }
+
+            _packet_params.modem_type = MODEM_LORA;
+
+            if ((_mod_params.params.lora.spreading_factor == LORA_SF5)
+                || (_mod_params.params.lora.spreading_factor == LORA_SF6))
+            {
+                if (preamble_len < 12)
+                {
+                    _packet_params.params.lora.preamble_length = 12;
+                }
+                else
+                {
+                    _packet_params.params.lora.preamble_length = preamble_len;
+                }
+            }
+            else
+            {
+                _packet_params.params.lora.preamble_length = preamble_len;
+            }
+
+            _packet_params.params.lora.header_type = (lora_pkt_length_t) fix_len;
+            _packet_params.params.lora.payload_length = max_payload_len;
+            _packet_params.params.lora.crc_mode = (lora_crc_mode_t) crc_on;
+            _packet_params.params.lora.invert_IQ = (lora_IQ_mode_t) iq_inverted;
+
+            set_modem(MODEM_LORA);
+
+            if (_reception_mode == RECEPTION_MODE_CONTINUOUS)
+            {
+                _rx_timeout = 0xFFFFFFFF;
+            }
+            else
+            {
+                _rx_timeout = 0x00000000;
+            }
+
+            break;
+        }
+
+        default:
+            break;
+    }
+}
+
+void STM32WL_LoRaRadio::configure_dio_irq(uint16_t irq_mask, uint16_t dio1_mask,
+                                          uint16_t dio2_mask, uint16_t dio3_mask)
+{
+    uint8_t buf[8];
+
+    buf[0] = (uint8_t)((irq_mask >> 8) & 0x00FF);
+    buf[1] = (uint8_t)(irq_mask & 0x00FF);
+    buf[2] = (uint8_t)((dio1_mask >> 8) & 0x00FF);
+    buf[3] = (uint8_t)(dio1_mask & 0x00FF);
+    buf[4] = (uint8_t)((dio2_mask >> 8) & 0x00FF);
+    buf[5] = (uint8_t)(dio2_mask & 0x00FF);
+    buf[6] = (uint8_t)((dio3_mask >> 8) & 0x00FF);
+    buf[7] = (uint8_t)(dio3_mask & 0x00FF);
+
+    write_opmode_command((uint8_t) RADIO_CFG_DIOIRQ, buf, 8);
+}
+
+void STM32WL_LoRaRadio::send(uint8_t *buffer, uint8_t size)
+{
+    set_tx_power(_tx_power);
+    configure_dio_irq(IRQ_TX_DONE | IRQ_RX_TX_TIMEOUT,
+                      IRQ_TX_DONE | IRQ_RX_TX_TIMEOUT,
+                      IRQ_RADIO_NONE,
+                      IRQ_RADIO_NONE);
+
+    /* ST_WORKAROUND_BEGIN : Set the debug pin and update the radio switch */
+    /* Set DBG pin */
+//    DBG_GPIO_RADIO_TX(SET);
+//  STM32WL_LoRaRadio::setTXPin(1);
```

Comment:
to remove ?

Reply:
I still have to find a way to set these debug pins (for now, static parameter compil issue).
I keep them as comments until resolved